ui/src/app/pages/login/account-creation/account-creation.tsx

```diff
@@ -0,0 +1,665 @@
+    this.usernameCheckTimeout = setTimeout(() => {
+      if (!username.trim()) {
+        this.setState({usernameCheckInProgress: false});
+        return;
+      }
+      profileApi().isUsernameTaken(username)
+        .then((body) => {
+          this.setState({usernameCheckInProgress: false, usernameConflictError: body.isTaken});
+        })
+        .catch((error) => {
+          console.log(error);
+          this.setState({usernameCheckInProgress: false});
+        });
+    }, 300);
+  }
+
+  updateProfile(attribute: string, value: string) {
+    if (attribute === 'contactEmail') {
+      this.setState({invalidEmail: false});
+    }
+    const newProfile = this.state.profile;
+    newProfile[attribute] = value;
+    this.setState(({profile}) => ({profile: fp.set(attribute, value, profile)}));
+  }
+
+  // The strings will be changed to ENUM value once survey page is done
+  updateAffiliationRoles(affiliation) {
+    this.setState({showAffiliationRole: false, showAffiliationOther: false,
+      affiliation: affiliation});
+    if (affiliation === 'industry') {
+      this.setState({rolesOptions: Options.industryRole, showAffiliationRole: true});
+    } else if (affiliation === 'educationalInstitution') {
+      this.setState({rolesOptions: Options.eductionRole, showAffiliationRole: true});
+    } else if (affiliation === 'freeText') {
+      this.setState({showAffiliationOther: true});
+      return;
+    }
+    this.selectAffiliationRoles(this.state.affiliationRole);
+  }
+
+  selectAffiliationRoles(role) {
+    this.setState({affiliationRole: role});
+    if (role === 'freeText') {
+      this.setState({showAffiliationOther: true});
+    } else {
+      this.setState({showAffiliationOther: false});
+    }
+  }
+
+  // This will be deleted once enableAccountPages is set to true for prod
+  validate() {
+    const {profile} = this.state;
+    const requiredFields =
+      [profile.givenName, profile.familyName, profile.username, profile.contactEmail,
+        profile.currentPosition, profile.organization, profile.areaOfResearch];
+    if (requiredFields.some(isBlank)) {
+      this.setState({showAllFieldsRequiredError: true});
+      return;
+    }
+  }
+
+  validateAccountCreation() {
+    const {profile: {
+      givenName, familyName, contactEmail, username
+    },
+      address: {
+        streetAddress1, city, state, zipcode, country
+      },
+      affiliation, affiliationRole, institutionName,
+      institutionRole
+    } = this.state;
+    const errors = validate({
+      givenName, familyName, streetAddress1, city, state, zipcode, country,
+      contactEmail, username, affiliation, affiliationRole, institutionName, institutionRole
+    }, {
+      givenName: {
+        presence: {allowEmpty: false}
+      },
+      familyName: {
+        presence: {allowEmpty: false}
+      },
+      contactEmail: {
+        presence: {allowEmpty: false},
+        format: {
+          pattern: /^[a-zA-Z0-9._-]+@[a-zA-Z0-9.-]+\.[a-zA-Z]{2,4}$/,
```

Comment:
Could you use the same regex pattern on line 190 for email validation?

Reply:
Its the same regex, the other is going to be deleted once we have the enable flag set to true